backend/app/services/db/rating_manager.py

```diff
@@ -0,0 +1,187 @@
+from datetime import datetime
+
+from .db_connect import connect
+from .user_manager import get_user
+
+
+def rate_profile(rater_name: str, rated_name: str, rating: int):
+    """Rates an existing profile.
+
+    Args:
+        rater_name (str): The username of the rating user.
+        rated_name (str): The username of the user receiving a rating.
+        rating (int): The rating received by the receiving user. Ranges from 1 to 5.
+
+    Returns:
+        True if the function succeeds, else False.
+    """
+
+    with connect() as conn:
+        cursor = conn.cursor()
+
+        # Ensure users cannot rate themselves
+        if rater_name == rated_name:
+            return False
+
+        # Check if users exist
+        userA_details = get_user(rater_name, is_username=True)
+        userB_details = get_user(rated_name, is_username=True)
+
+        if not userA_details or not userB_details:
+            return False
+
+        # Check if a rating already exists
+        cursor.execute(f'''
+                       SELECT * FROM ratings
+                       WHERE rater_name = \"{rater_name}\" AND rated_name = \"{rated_name}\"
+                       ''')
+
+        # If a rating already exists, update the rating.
+        if cursor.fetchall():
+            update_rating(rater_name, rated_name, rating)
+
+            return True
+
+        query = '''
+        INSERT INTO ratings (rater_name, rated_name, rating, rated_date)
+        VALUES (%s, %s, %s, %s)
+        '''
+
+        values = (
+            userA_details['username'],
+            userB_details['username'],
+            rating,
+            datetime.now()
+        )
+
+        # Insert ratings into ratings table
+        cursor.execute(query, values)
+        conn.commit()
+
+    return True
+
+
+def update_rating(rater_name: str, rated_name: str, rating: int):
+    """Updates the information of a user in the database.
+
+    Args:
+        rater_name (str): The username of the rating user.
+        rated_name (str): The username of the user receiving a rating.
+        rating (int): The rating received by the receiving user. Ranges from 1 to 5.
+
+    Returns:
+        True if the function succeeds, else False.
+    """
+
+    with connect() as conn:
+        cursor = conn.cursor()
+
+        # Check if users exist
+        existing = get_specific_rating(rater_name, rated_name)
+
+        if not existing:
+            return False
+
+        query = '''
+        UPDATE ratings SET
+        rating = COALESCE(%s, rating)
+        WHERE rater_name = %s AND rated_name = %s
+        '''
+
+        values = (
+            rating,
+            rater_name,
+            rated_name
+        )
+
+        cursor.execute(query, values)
+        conn.commit()
+
+    return True
+
+
+def get_rating(username: str, is_rater: bool = True):
```

Comment:
Should be get_ratings(). Also, this function probably isn't needed for a rater as get_avg_rating can take care of that functionality.